### PR TITLE
Please review the Use Case and rest api 

### PR DESCRIPTION
**Pull request by dandanmylady**
@crook @jay-tsu, Please review the use case
@crook, please review the rest api which is required by Ethan
@crook, the comments about DB is also included in this file, please have a check

### CODE REVIEW

File: documentation/Use Case draft version.txt [hunk 1]
```diff
@@ -1,56 +1,56 @@
...
+    12. Remove 'isequiped' from gpssensor
+    13. The type of status should be Int not boolean
+    14. Add hight to gpsdata table
+   
+
+
+    
+User Cases from Customer view:
+1. user login from wechat(wechat id) or webpage(user, pws)
+    *a) if there isn't user mapping for the wechat id, prompt register page
```
Comment (crook) [inline documentation/Use Case draft version.txt:21]: how to mapp wechat id with user id in db?

File: documentation/Use Case draft version.txt [hunk 2]
```diff
@@ -1,56 +1,56 @@
...
-    
-3. Admin can click one specify equip to view  detailed info 
-show last one week gps data?
-
+DB design Issue:
+    1. Short the id name
+    2. Consider to adopt MongoDB
+    3. Add Admin table
+    4. Add Company table
+    5. Add Role Table
```
Comment (crook) [inline documentation/Use Case draft version.txt:6]: As for admin/Role table, so far, the django auth system has already the following 3 tables, is it enough?

```
Users
Permissions: Binary (yes/no) flags designating whether a user may perform a certain task.
Groups: A generic way of applying labels and permissions to more than one user.
```

File: documentation/Use Case draft version.txt [hunk 3]
```diff
@@ -1,56 +1,56 @@
...
+    6. Add Vendor Table
+    8. Add SN/Disc to Equip
```
Comment (crook) [inline documentation/Use Case draft version.txt:8]: what's Disc?

File: documentation/Use Case draft version.txt [hunk 4]
```diff
@@ -1,56 +1,56 @@
...
+3. Brower Equps, Place Order and Apply. Then user can review order list
+
+
+User Cases from Admin's view:
+1. Login 
+REST API
+{
+    user: xxx,
+    password: xxx
+}
```
Comment (crook) [inline documentation/Use Case draft version.txt:35]: @Ethan, is this user/password auth enough from your side? do you need oauth?

File: documentation/Use Case draft version.txt [hunk 5]
```diff
@@ -1,56 +1,56 @@
...
+            gps_status: working?,
+            gps_latest_position_x: 132,
+            gps_latest_position_y: 23, 
+        }
+    }
+    
+    2. filter for equip: filter by model, status, illegual?
+    
+3. Admin can click one specify equip to view  detailed info 
+show last one week gps data?
```
Comment (crook) [inline documentation/Use Case draft version.txt:54]: The REST API will support filter query ASAP

File: documentation/Use Case draft version.txt (hunk 2 above)
Comment (dandanmylady) [reply documentation/Use Case draft version.txt:6]: it's ok to take advantage of Django framework. But we need to point it out when we design our DB.
One thing to clarify is that the "admin" is the owner of equip renting company, like Jay's wife. :-)

File: documentation/Use Case draft version.txt (hunk 3 above)
Comment (dandanmylady) [reply documentation/Use Case draft version.txt:8]: description

File: documentation/Use Case draft version.txt (hunk 1 above)
Comment (dandanmylady) [reply documentation/Use Case draft version.txt:21]: @ethanxuyingjun, as we discussed WeChat id is mapped to the user id when she/he first time login from the WeChat, right?

File: documentation/Use Case draft version.txt (hunk 4 above)
Comment (crook) [reply documentation/Use Case draft version.txt:35]: @ethanxuyingjun

**Comment by crook** [comment]
@dandanmylady, few  comments:
1. how about use 'docs' foldername?
2. plese use 'git stash' to merge all your commits into one final commit after review pass

File: db/jksndb.sql [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+  CONSTRAINT `id`
+    FOREIGN KEY ()
+    REFERENCES `jksndb`.`category` ()
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `jksndb`.`order`
```
Comment (crook) [inline db/jksndb.sql:116]: we need a filed to identify this order belongs to whom. call 'buyer'  or 'owner'? 
I.e.  The admin  may help end user make one order.   In this case,  shall need 2 field to seprate?

File: db/jksndb.sql [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `jksndb`.`order` (
+  `id` INT NOT NULL,
+  `signtime` DATETIME NULL,
+  `starttime` DATETIME NULL,
+  `endtime` DATETIME NULL,
+  `duration` INT NULL,
+  `money` INT NULL,
+  PRIMARY KEY (`id`),
```
Comment (crook) [inline db/jksndb.sql:125]: suggest to add more 'stauts' to save the order stauts. that would be userful for search/filter

The Equip model is same

File: db/jksndb.sql [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `jksndb`.`category` (
+  `id` INT NOT NULL,
+  `discription` VARCHAR(45) NULL,
+  PRIMARY KEY (`id`))
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `jksndb`.`equip`
```
Comment (crook) [inline db/jksndb.sql:94]: Missing File/Video fiedl to het user to upload sth to show their device

File: db/jksndb.sql [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `jksndb`.`equip` (
+  `id` INT NOT NULL,
+  `model` VARCHAR(45) NULL,
+  `sn` VARCHAR(45) NULL,
+  `discription` VARCHAR(45) NULL,
```
Comment (crook) [inline db/jksndb.sql:100]: description

File: db/jksndb.sql [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+
+-- -----------------------------------------------------
+-- Table `jksndb`.`gpsdata`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `jksndb`.`gpsdata` (
+  `id` INT NOT NULL,
+  `time` DATETIME NULL,
+  `x` FLOAT NULL,
+  `y` FLOAT NULL,
+  `hight` FLOAT NULL,
```
Comment (crook) [inline db/jksndb.sql:174]: should be height.

**Pull request merged**